R/derive_merged.R

```diff
@@ -173,136 +180,188 @@
 #'
 #' @export
 #'
-#' @examples
+#' @examplesx
+#'
+#' @caption Basic merge of a full dataset
+#' @info Merge all demographic variables onto a vital signs dataset.
+#'
+#' - The variable `DOMAIN` exists in both datasets so note the use of
+#'   `select(dm, -DOMAIN)` in the `dataset_add` argument.
+#' @code
+#' library(tibble)
 #' library(dplyr, warn.conflicts = FALSE)
 #' vs <- tribble(
-#'   ~STUDYID,  ~DOMAIN,  ~USUBJID, ~VSTESTCD,      ~VISIT, ~VSSTRESN, ~VSSTRESU,       ~VSDTC,
-#'   "PILOT01",    "VS", "01-1302",  "HEIGHT", "SCREENING",     177.8,      "cm", "2013-08-20",
-#'   "PILOT01",    "VS", "01-1302",  "WEIGHT", "SCREENING",     81.19,      "kg", "2013-08-20",
-#'   "PILOT01",    "VS", "01-1302",  "WEIGHT",  "BASELINE",      82.1,      "kg", "2013-08-29",
-#'   "PILOT01",    "VS", "01-1302",  "WEIGHT",    "WEEK 2",     81.19,      "kg", "2013-09-15",
-#'   "PILOT01",    "VS", "01-1302",  "WEIGHT",    "WEEK 4",     82.56,      "kg", "2013-09-24",
-#'   "PILOT01",    "VS", "01-1302",  "WEIGHT",    "WEEK 6",     80.74,      "kg", "2013-10-08",
-#'   "PILOT01",    "VS", "01-1302",  "WEIGHT",    "WEEK 8",      82.1,      "kg", "2013-10-22",
-#'   "PILOT01",    "VS", "01-1302",  "WEIGHT",   "WEEK 12",      82.1,      "kg", "2013-11-05",
-#'   "PILOT01",    "VS", "17-1344",  "HEIGHT", "SCREENING",     163.5,      "cm", "2014-01-01",
-#'   "PILOT01",    "VS", "17-1344",  "WEIGHT", "SCREENING",     58.06,      "kg", "2014-01-01",
-#'   "PILOT01",    "VS", "17-1344",  "WEIGHT",  "BASELINE",     58.06,      "kg", "2014-01-11",
-#'   "PILOT01",    "VS", "17-1344",  "WEIGHT",    "WEEK 2",     58.97,      "kg", "2014-01-24",
-#'   "PILOT01",    "VS", "17-1344",  "WEIGHT",    "WEEK 4",     57.97,      "kg", "2014-02-07",
-#'   "PILOT01",    "VS", "17-1344",  "WEIGHT",    "WEEK 6",     58.97,      "kg", "2014-02-19",
-#'   "PILOT01",    "VS", "17-1344",  "WEIGHT",    "WEEK 8",     57.79,      "kg", "2014-03-14"
-#' )
+#'   ~DOMAIN,  ~USUBJID, ~VSTESTCD, ~VISIT,      ~VSSTRESN, ~VSDTC,
+#'   "VS",     "01",     "HEIGHT",  "SCREENING",     178.0, "2013-08-20",
+#'   "VS",     "01",     "WEIGHT",  "SCREENING",      81.9, "2013-08-20",
+#'   "VS",     "01",     "WEIGHT",  "BASELINE",       82.1, "2013-08-29",
+#'   "VS",     "01",     "WEIGHT",  "WEEK 2",         81.9, "2013-09-15",
+#'   "VS",     "01",     "WEIGHT",  "WEEK 4",         82.6, "2013-09-24",
+#'   "VS",     "02",     "WEIGHT",  "BASELINE",       58.6, "2014-01-11"
+#' ) %>%
```

Comment:
Space permitting, what about adding a 3rd patient here with an example of NAs in the sort order variable? Then this patient can be referenced in the second example because the week 2 weight will be picked instead of the Week 4
```
  "VS",     "03",     "WEIGHT",  "BASELINE",       80.1, "2015-10-14",
  "VS",     "03",     "WEIGHT",  "WEEK 2",         81.3, NA_character_,
  "VS",     "03",     "WEIGHT",  "WEEK 4",         80.7, "2015-11-15"
```

Reply:
Now seeing that you may then need to add a subject 04 to DM to ensure other examples make sense, or similar.